lib/Dialect/FIRRTL/Transforms/LowerClasses.cpp

```diff
@@ -574,44 +741,8 @@ std::unique_ptr<mlir::Pass> circt::firrtl::createLowerClassesPass() {
 bool LowerClassesPass::shouldCreateClass(FModuleLike moduleLike) {
   // Memoize calls using the extra state.
   auto it = shouldCreateClassMemo.find(moduleLike);
-  if (it != shouldCreateClassMemo.end())
-    return it->second;
-
-  if (isa<firrtl::ClassLike>(moduleLike.getOperation())) {
-    shouldCreateClassMemo.insert({moduleLike, true});
-    return true;
-  }
-
-  // Always create a class for public modules.
-  if (moduleLike.isPublic()) {
-    shouldCreateClassMemo.insert({moduleLike, true});
-    return true;
-  }
-
-  // Create a class for modules with property ports.
-  bool hasClassPorts = llvm::any_of(moduleLike.getPorts(), [](PortInfo port) {
-    return isa<PropertyType>(port.type);
-  });
-
-  if (hasClassPorts) {
-    shouldCreateClassMemo.insert({moduleLike, true});
-    return true;
-  }
-
-  // Create a class for modules that instantiate classes or modules with
-  // property ports.
-  for (auto op :
-       moduleLike.getOperation()->getRegion(0).getOps<FInstanceLike>()) {
-    for (auto result : op->getResults()) {
-      if (type_isa<PropertyType>(result.getType())) {
-        shouldCreateClassMemo.insert({moduleLike, true});
-        return true;
-      }
-    }
-  }
-
-  shouldCreateClassMemo.insert({moduleLike, false});
-  return false;
+  assert(it != shouldCreateClassMemo.end() && "it must be computed beforehand");
```

Comment:
FWIW this is what `at(moduleLike)` does (find, assert, return it->second).

Reply:
Thanks!